gear/gear/auth.py

```diff
@@ -12,19 +12,34 @@
 from hailtop.auth import async_get_userinfo
 from hailtop.config import get_deploy_config
 
+from .time_limited_max_size_cache import TimeLimitedMaxSizeCache
+
 log = logging.getLogger('gear.auth')
 
 deploy_config = get_deploy_config()
 
 BEARER = 'Bearer '
 
+TEN_SECONDS_IN_NANOSECONDS = int(1e10)
+
+USERDATA_CACHE = None
```

Comment:
Hmm. Ok, I understand why you lazily initialize but I’m just generally a bit hesitant with mutation-based initialization like this. How bad is it if we have to explicitly initialize a class that exposes this file’s functions as an interface? And then annotations are like @authclient.web_authenticated_user etc.

Reply:
Ya I don't like this either, but I needed some way to get the client session into the lookup function without it being the key. I'm not sure I see how using the class would help here because we use the decorators at the top-level and we only have the client session once we've entered main?

Maybe this is orthogonal to your point of having a class, which I'm not at all opposed to, but I suppose I could make the key a tuple of session_id and session, and get it to the lookup function that way. Both that and the current approach feel a little janky.

Is there one you prefer?